x-pack/filebeat/input/azureblobstorage/input_test.go

```diff
@@ -0,0 +1,328 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package azureblobstorage
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/sync/errgroup"
+
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	beattest "github.com/elastic/beats/v7/libbeat/publisher/testing"
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/azureblobstorage/mock"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+const (
+	beatsContainer  = "beatscontainer"
+	beatsContainer2 = "beatscontainer2"
+)
+
+func Test_StorageClient(t *testing.T) {
+	tests := []struct {
+		name            string
+		baseConfig      map[string]interface{}
+		mockHandler     func() http.Handler
+		expected        map[string]bool
+		isError         error
+		unexpectedError error
```

Comment:
Not for now: I would call these `wantErr` and `waitErr` to be more clearly named. (`isFoo` suggests a boolean and `unexpectedFoo` suggests something that is unexpected).